atom/browser/api/atom_api_web_contents.h

```diff
@@ -462,6 +464,41 @@ class WebContents : public mate::TrackableObject<WebContents>,
 #endif
 
  private:
+  // Handles IPC messages from a specific RenderFrame. Created lazily.
+  // TODO(nornagon): this probably shouldn't be tied to the api::WebContents.
+  // Message/MessageSync are api::WebContents-specific but it's also totally
+  // plausible that there would be IPC messages that could be handled by the
+  // browser process that don't require an api::WebContents to exist.
+  class IPCHandler : public electron_api::mojom::ElectronBrowser,
```

Comment:
yeah this seems to be a combination of `OnMessageReceived` and `FrameDispatchHelper`. Maybe we can have interfaces specific to `api::WebContents` declared separately from the generic ones, that way we can override them directly without having an `IPCHandler` class ?

Reply:
Not quite sure what you're saying; that `api::WebContents` should derive from `mojom::ElectronBrowser`?

Reply:
I tried folding the `mojom::ElectronBrowser` implementation up into `api::WebContents`, but ran into a snag, which is that `content::WebContentsFrameBindingSet` must be initialized in the constructor, but not all constructors of `api::WebContents` have a relevant `content::WebContents` pointer we could pass in.

I think this is a weakness in the design of `api::WebContents`, but I don't plan to fix that here.